pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -285,6 +288,25 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 
 	pLog.Debug("install job still running")
 
+	if time.Since(job.CreationTimestamp.Time) > podStatusCheckDelay {
+		installPod, err := r.getInstallPod(job, pLog)
+		if err != nil {
+			pLog.WithError(err).Error("could not get install pod")
+			return r.transitionStage(instance, hivev1.ClusterProvisionStageFailed, "InstallPodMissing", err.Error(), pLog)
+		}
+
+		if installPod.Status.Phase == "Pending" {
+			pLog.WithField("pod", installPod.Name).Error("install pod is stuck")
+			if err := r.setCondition(instance, hivev1.InstallPodStuckCondition, corev1.ConditionTrue, "PodInPendingPhase", "pod is in pending phase", controllerutils.UpdateConditionIfReasonOrMessageChange, pLog); err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
+		}
+		if err := r.setCondition(instance, hivev1.InstallPodStuckCondition, corev1.ConditionFalse, "PodInRunningPhase", "pod is in running phase", controllerutils.UpdateConditionAlways, pLog); err != nil {
```

Comment:
Since `setCondition` is not doing any change checks, this will do a status update on the ClusterProvision even if nothing has changed.